File: programming/2.3/main.cpp
#include <iostream>
#include <cmath>
#include <string> 


using namespace std;

string naturalNumberView[] = {
  "zero",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eigth",
  "nine",
  "ten"
};

string dozensNumberView[] = {
  "",
  "Ten",
  "Twenty",
  "Thirty",
  "Fourty",
  "Fifty",
  "Sixty",
  "Seventy",
  "Eighty",
  "Ninety"
};

string toTwenty[] = {
  "",
  "eleven",
  "twelve",
  "thirteen",
  "fuorteen",
  "fifteen",
  "sixteen",
  "seventenn",
  "eightenn",
  "ninetenn"
};

int randint(int min, int max) {
  return rand() % (max + 1 - min) + min;
}

string parseNaturalNumbers(int number) {
  // parse natural numbers 
  if (number % 10) {
    return naturalNumberView[number % 10] + ' ';
  }
  return "";
}

string parseNumberYear(int number) {
  string outString;
  int toHundred = floor(number / 10);

  // parse dozens number
  if (number >= 20) {
    outString += dozensNumberView[toHundred] + ' ';
    outString += parseNaturalNumbers(number);
  }
  else if (number <= 20 && number >= 11) {
    outString += toTwenty[number % 10] + ' ';
  }
  else if (number <= 10) {
    outString += naturalNumberView[number] + ' ';
  }
  
  return outString;
}

int main() {
  srand(time(NULL));
  
  int yearNumber = randint(20, 69);

  cout << "Year number: " << yearNumber << endl;

  cout << parseNumberYear(yearNumber) << endl;
 
  return 0;
}

File: programming/2.3/main.py
from ast import parse
from random import randint 


natural_number_view = [
  'zero',
  'one',
  'two',
  'three',
  'four',
  'five',
  'six',
  'seven',
  'eight',
  'nine',
  'ten',
]

to_twenty_number_view = [
  '',
  'eleven',
  'twelve',
  'thirteen',
  'fuorteen',
  'fifteen',
  'sixteen',
  'seventenn',
  'eightenn',
  'ninetenn',
]

dozens_number_view = [
  '',
  'ten',
  'twenty',
  'thirty',
  'fourty',
  'fifty',
  'sixty',
  'seventy',
  'eighty',
  'ninety',
]

def parse_natural_numbers(number: int) -> str:
  if (number % 10 <= 10 and number % 10):
    return natural_number_view[number % 10]
  
  return ''


def parse_to_twenty_numbers(number: int) -> str:
  if (number < 20 and number > 10):
    return to_twenty_number_view[number % 10]

  return ''


def parse_dozens_numbers(number: int) -> str:
  dozens_number = number // 10

  if (dozens_number_view != 1):
    return dozens_number_view[dozens_number]

  return ''


def parse_number_year(number: int) -> str:
  out_string = []


  if number <= 10:
    out_string.append(parse_natural_numbers(number))

  elif number <= 11 and number >= 20:
    out_string.append(parse_to_twenty_numbers(number))

  elif number >=20:
    out_string.append(parse_dozens_numbers(number))
    out_string.append(parse_natural_numbers(number))

  out_string[0] = out_string[0].title()

  return ' '.join(out_string)


def main():
  yearNumber = randint(20, 69)

  print('Year number:', yearNumber)

  print(parse_number_year(yearNumber))


if __name__ == '__main__':
  main()
